cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2146,6 +2147,182 @@ class TakeMetaFunction : public MetaFunction {
   }
 };
 
+// ----------------------------------------------------------------------
+// DropNull Implementation
+
+Status GetDropNullFilter(const Array& values, MemoryPool* memory_pool,
+                         std::shared_ptr<arrow::BooleanArray>* out_array) {
+  auto bitmap_buffer = values.null_bitmap();
+  *out_array = std::make_shared<BooleanArray>(values.length(), bitmap_buffer, nullptr,
+                                              values.null_count(), values.offset());
```

Comment:
```suggestion
                                              0, values.offset());
```
Once the validity buffer is promoted to a data buffer, we should consider it as a non-null array.   `values.null_count() != 0`, so, we can simply pass 0 here.